### PR TITLE
Add a Chart for Risk Advisor

### PR DESCRIPTION
**Pull request by pposkrobko**
Hi,

We are a group of Computer Science students from University of Warsaw and we made Risk Advisor module for Kubernetes as our Bachelor's thesis in cooperation with @crassirostris from Google Warsaw.

The source code can be found here: https://github.com/Prytu/risk-advisor

Cheers,
Pawel

### CODE REVIEW

File: stable/risk-advisor/values.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  repository: pposkrobko/risk-advisor
+  tag: v1.0.0
+  pullPolicy: IfNotPresent
+service:
+  name: risk-advisor-service
+  type: NodePort
+  externalPort: 11111
+  internalPort: 9997
+resources:
+  limits:
```
Comment (unguiculus) [inline stable/risk-advisor/values.yaml]: Don't provide defaults for resources

File: stable/risk-advisor/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
```
Comment (unguiculus) [inline stable/risk-advisor/templates/deployment.yaml]: See best practices for standard labels and apply them everywhere:
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/labels.md

Use `{{ template "name" . }}` for the `app` label.

File: stable/risk-advisor/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# risk-advisor
```
Comment (unguiculus) [inline stable/risk-advisor/README.md:1]: Please follow the standard readme style. See e. g. https://github.com/kubernetes/charts/blob/master/stable/nginx-ingress/README.md

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by unguiculus** [comment]
Closing this due to inactivity. Feel free to re-open if you decide to continue working on it.

**Comment by pposkrobko** [comment]
@unguiculus, I'd like to continue working on it and I added suggested fixes to my fork, however can't see a way to reopen this PR by myself, could you please reopen it? Or do you want me to create new PR?

**Comment by pposkrobko** [comment]
thanks, I see new commits have been added automatically, I hope PR is ok now, please verify

File: stable/risk-advisor/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+name: risk-advisor
+description: Risk Advisor add-on module for Kubernetes
+version: 1.0.0
+sources:
+- https://github.com/Prytu/risk-advisor/
+icon: https://raw.githubusercontent.com/Prytu/risk-advisor/master/logo.svg
```
Comment (unguiculus) [inline stable/risk-advisor/Chart.yaml]: Maintainer is missing. User Github username.

File: stable/risk-advisor/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+## Uninstalling the Chart
+
+To uninstall/delete the `my-release` deployment:
+
+```console
+$ helm delete my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
```
Comment (unguiculus) [inline stable/risk-advisor/README.md:35]: Add a configuration section.

File: stable/risk-advisor/templates/NOTES.txt [hunk 6]
```diff
@@ -0,0 +1,2 @@
+1. Check how the cluster state would change after adding provided pods by running these command:
+	curl -XPOST SERVICE_URL:11111/advise -H "Content-type: application/json" -d @PATH_TO_JSON_WITH_PODS
```
Comment (unguiculus) [inline stable/risk-advisor/templates/NOTES.txt]: This is very little information. URL/port should not be hard-coded.

Here's a good example: https://github.com/kubernetes/charts/blob/master/stable/nginx-ingress/templates/NOTES.txt

File: stable/risk-advisor/templates/service.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    heritage: "{{ .Release.Service }}"
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+  - port: {{ .Values.service.externalPort }}
+    targetPort: {{ .Values.service.internalPort }}
+    protocol: TCP
+    name: {{ .Values.service.name }}
+  selector:
+    app: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/risk-advisor/templates/service.yaml]: Selector should be this:

```yaml
app: "{{ template "name" . }}"
release: "{{ .Release.Name }}"
```

**Review by unguiculus — CHANGES_REQUESTED** [review]
A few more things.

**Comment by pposkrobko** [comment]
@unguiculus thanks for pointing these things out, I have done all the changes you've requested

File: stable/risk-advisor/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    app: "{{ template "name" . }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
```
Comment (unguiculus) [inline stable/risk-advisor/templates/deployment.yaml:14]: Only keep `app` and `release` label here.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/risk-advisor/templates/deployment.yaml (hunk 8 above)
Comment (pposkrobko) [reply stable/risk-advisor/templates/deployment.yaml:14]: done

**Review by unguiculus — APPROVED** [review]

**Comment by unguiculus** [comment]
@k8s-bot ok to test

**Pull request merged**